training_record/views.py

```diff
@@ -91,6 +94,53 @@ def reviewed(self, request):
         )
         return self._get_paginated_response(queryset)
 
+    @decorators.action(detail=False, methods=['GET'],
+                       url_path='search')
+    def search(self, request):
```

Comment:
I think you can customize [APIs]((https://django-filter.readthedocs.io/en/master/guide/usage.html#customize-filtering-with-filter-method)) provided by `django-filter`, then this method is no longer necessary.